src/compiler/transformers/es6.ts

```diff
@@ -1684,6 +1686,119 @@ namespace ts {
             return convertIterationStatementBodyIfNecessary(node);
         }
 
+        // // TODO(rbuckton): Switch to using __values helper for for..of?

```

Comment:
I mean, it looks pretty nice/clean - though I imagine the difference in capabilities/output size/runtime speed matter more. In any case, this block should either be un-commented and used to replace the other visitForOfStatement or removed. ;)

Though, if it's not strictly required for downlevel generators, I imagine it needs to be moved into a followup PR.